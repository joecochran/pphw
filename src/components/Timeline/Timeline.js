import React from 'react';
import './Timeline.scss';

function Timeline() {
  return (
    <ul className="c-timeline">
      <li className="c-timeline__month">2018</li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month">Q1 2019</li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month">Q2 2019</li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month">Q3 2019</li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month"></li>
      <li className="c-timeline__month">Q4 2019</li>
      <li className="c-timeline__month"></li>
    </ul>
  )
}

export default Timeline;
