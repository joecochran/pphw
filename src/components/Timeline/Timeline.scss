.c-timeline {
  display: grid;
  grid-template-columns: repeat(13, 100px);
  list-style: none;
  margin: 0;
  padding: 17px 0 0 18px;
  margin-bottom: 16px;

  &__month {
    background-image: url('data:image/svg+xml;charset=US-ASCII,%3Csvg%20width%3D%224%22%20height%3D%224%22%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%3E%0A%20%20%3Crect%20y%3D%2224%22%20width%3D%224%22%20height%3D%224%22%20rx%3D%222%22%20transform%3D%22translate%280%20-24%29%22%20fill%3D%22%23C1C4C7%22%20fill-rule%3D%22nonzero%22/%3E%0A%3C/svg%3E%0A');
    background-repeat: no-repeat;
    height: 28px;
    background-position: bottom left;
    font-weight: bold;
    color: #5e6266;
  }
}
