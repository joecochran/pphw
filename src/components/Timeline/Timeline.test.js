import React from 'react';
import Timeline from './Timeline';
import { shallow } from 'enzyme';

describe('<Timeline>', () => {
  const wrapper = shallow(<Timeline />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
