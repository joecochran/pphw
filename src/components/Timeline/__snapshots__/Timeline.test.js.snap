// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeline />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="c-timeline__month"
        >
          2018
        </li>,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        >
          Q1 2019
        </li>,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        >
          Q2 2019
        </li>,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        >
          Q3 2019
        </li>,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        />,
        <li
          className="c-timeline__month"
        >
          Q4 2019
        </li>,
        <li
          className="c-timeline__month"
        />,
      ],
      "className": "c-timeline",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018",
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": "2018",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Q1 2019",
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": "Q1 2019",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Q2 2019",
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": "Q2 2019",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Q3 2019",
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": "Q3 2019",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Q4 2019",
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": "Q4 2019",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-timeline__month",
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="c-timeline__month"
          >
            2018
          </li>,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          >
            Q1 2019
          </li>,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          >
            Q2 2019
          </li>,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          >
            Q3 2019
          </li>,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          />,
          <li
            className="c-timeline__month"
          >
            Q4 2019
          </li>,
          <li
            className="c-timeline__month"
          />,
        ],
        "className": "c-timeline",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018",
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": "2018",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Q1 2019",
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": "Q1 2019",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Q2 2019",
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": "Q2 2019",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Q3 2019",
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": "Q3 2019",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Q4 2019",
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": "Q4 2019",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-timeline__month",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
